### PR TITLE
Show ride and vehicle names in research list

### PR DESCRIPTION
**Pull request by Gymnasiast**
Fixes one of the issues in #7324.

forest frontiers 2018-03-21 13-51-16

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
This is a great improvement. However, I think it'd be even better to columnise the list, like the object selection window.

File: src/openrct2-ui/windows/EditorInventionsList.cpp [hunk 1]
```diff
@@ -780,14 +784,30 @@ static void window_editor_inventions_list_scrollpaint(rct_window *w, rct_drawpix
...
 
-        format_string(ptr, 256, stringId, nullptr);
+        if (researchItem->type == RESEARCH_ENTRY_TYPE_RIDE && !RideGroupManager::RideTypeIsIndependent(researchItem->baseRideType))
+        {
+            rct_string_id rideGroupName = get_ride_naming(researchItem->baseRideType, get_ride_entry(researchItem->entryIndex)).name;
+            format_string(ptr, 256, rideGroupName, nullptr);
+            while (*ptr != (utf8)'\0')
+            {
+                ptr++;
+            }
```
Comment (IntelOrca) [inline src/openrct2-ui/windows/EditorInventionsList.cpp]: `std::strchr` does this for you.

Comment (Gymnasiast) [reply src/openrct2-ui/windows/EditorInventionsList.cpp]: I have a better version incoming.

**Comment by Gymnasiast** [comment]
@AaronVanGeffen I have updated the PR and the first post. I chose this approach because I find it slightly cleaner looking, especially considering there's also scenery in this list.

**Comment by AaronVanGeffen** [comment]
I appreciate the effort — thanks! This looks much better. Not too keen on the colons, though. I can see you put them there for rides with a subtype only, but I don't think it adds much in the tabular set-up.

**Comment by Gymnasiast** [comment]
The Shortcut keys window does use them, though. Personally, I find it slightly easier to read like this, especially if the name on the left is long.

**Pull request merged**